CompilerSource/ipc/Shader.fbs

```diff
@@ -0,0 +1,15 @@
+
+enum ShaderType : byte { GLSL, GLSLES, HLSL9, HLSL11 }
+
+table Shader {
+  name:string;
+  id:int;
+
+  type:ShaderType;
+  precompile:bool;

```

Comment:
This precompile boolean is a field I added to ENIGMA that does not exist in GMS. Do we want to keep this around and continue going in this direction or remove it? I think it's useful to allow control over this.